C#.md

```diff
@@ -0,0 +1,497 @@
+
+The more visible a piece of code is (and by extension - the farther away consumers might be), the more documentation is needed.
+
+### General Documentation Quality
+
+* Use [XMLDoc](http://msdn.microsoft.com/en-us/library/b2s063f7%28v=VS.100%29.aspx)-style comments.
+* Every public class or method **must** have a comment.
+* Don't document overriding methods unless you want to tell what's different in its behavior.
+* Documentation text should be written using complete sentences, full stops are not necessary.
+
+### Documenting a Class
+Documentation for a class may range from a single sentence to paragraphs with code examples. Documentation should serve to disambiguate any conceptual blanks in the API, and make it easier to quickly and **correctly** use your API. A thorough class doc usually has a one sentence summary and, if necessary, a more detailed explanation.
+
+```C#
+/// <summary>
+/// Implements a variable-size List that uses an array of objects to store the
+/// elements. A List has a capacity, which is the allocated length
+/// of the internal array. As elements are added to a List, the capacity
+/// of the List is automatically increased as required by reallocating the
+/// internal array.
+/// </summary>
+public class List<T> : IList<T>, System.Collections.IList, IReadOnlyList<T>
+{
+	...
+}
+```
+
+### Documenting a Method
+
+A method doc should tell what the method does. Depending on the argument types, it may also be important to document input format.
+
+```C#
+/// <summary>
+/// Adds two doubles and returns the result.
+/// </summary>
+/// <returns>
+/// The sum of two doubles.
+/// </returns>
+/// <exception cref="System.OverflowException">Thrown when one parameter is max 
+/// and the other is greater than zero.</exception>
+/// See <see cref="Math.Add(int, int)"/> to add integers.
+/// <param name="a">A double precision number.</param>
+/// <param name="b">A double precision number.</param>
+public static double Add(double a, double b)
+{
+	...
+}
+```
+
+### Inline Comments
+
+Inline comments should always be added when the intent or purpose of any code isn't completely explicit, but the code itself ought to be clear enough to follow logically.
+
+```C#
+private void EnsureCapacity(int min) 
+{
+	if (m_Items.Length < min) 
+	{
+		var newCapacity = m_Items.Length == 0 ? m_DefaultCapacity : m_Items.Length * 2;
+		// Allow the list to grow to maximum possible capacity (~2G elements) before encountering overflow.
+		// Note that this check works even when m_Items.Length overflowed thanks to the (uint) cast
+		if ((uint)newCapacity > Array.MaxArrayLength) newCapacity = Array.MaxArrayLength;
+		if (newCapacity < min) newCapacity = min;
+		Capacity = newCapacity;
+	}
+}
+```
+
+## General Code Quality
+
+**Single responsibility principle**
+
+Every module, class or method should have responsibility over a single part of the functionality provided by the software.

```

Comment:
Add a wiki link here plz. SRP is of utmost importance and it'd be better to link a detailed explanation of what it is.